dhis-2/dhis-web/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/AbstractCrudController.java

```diff
@@ -822,6 +854,37 @@ public void removeAsFavorite( @PathVariable( "uid" ) String pvUid, HttpServletRe
         webMessageService.send( WebMessageUtils.ok( message ), response, request );
     }
 
+    @RequestMapping( value = "/{uid}/subscriber", method = RequestMethod.DELETE )
+    @ResponseStatus( HttpStatus.OK )
+    public void unsubscribe( @PathVariable( "uid" ) String pvUid, HttpServletRequest request, HttpServletResponse response ) throws Exception
+    {
+        if ( !getSchema().isSubscribable() )
+        {
+            throw new WebMessageException( WebMessageUtils.conflict( "Objects of this class cannot be subscribed to" ) );
+        }
+
+        List<SubscribableObject> entity = (List<SubscribableObject>) getEntity( pvUid );
+
+        if ( entity.isEmpty() )
+        {
+            throw new WebMessageException( WebMessageUtils.notFound( getEntityClass(), pvUid ) );
+        }
+
+        SubscribableObject object = entity.get( 0 );
+        User user = currentUserService.getCurrentUser();
+
+        if ( user == null )
```

Comment:
This is not really necessary, you can't access the web api without a active security context

Reply:
Okay lets remove that, I have updated the favorites code to reflect it.

Reply:
Ok, I have just removed it from the subscribe/unsubscribe methods as well. I also updated the branch with the latest from master...